src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -183,7 +183,7 @@ private static BuildEnvironment TryFromMSBuildAssembly()
             var msBuildDll = Path.Combine(FileUtilities.GetFolderAbove(buildAssembly), "MSBuild.dll");
 
             // First check if we're in a VS installation
-            var environment = TryFromMSBuildAssemblyUnderVisualStudio(buildAssembly, msBuildExe);
```

Comment:
If we're calculating MSBuild.exe's path anyway, why pass through this assembly over that one? As far as I can tell, it just means we need to do a little more avoidable work later when it comes to calling GetMSBuildExeFromVsRoot.

Reply:
Yeah, looked like both callers could use just the path so switched to that (and renamed for clarity).

Reply:
As @ladipro pointed out this was wrong. We compute _a_ path to MSBuild but the right MSBuild.exe path may be elsewhere (in the amd64 folder), so we have to recompute it anyway.